### ISSUE TITLE
panic: runtime error: SIGSEGV on non existing directory

### ISSUE THREAD

**Issue by krhubert** [uber-go/zap#390]
Hi,

There is panic (trying to execute nil function) when there is the directory non exists.

```
package main

import (
	"log"

	"go.uber.org/zap"
	"go.uber.org/zap/zapcore"
)

func main() {
	cfg := zap.NewProductionConfig()
	cfg.EncoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
	cfg.Level.SetLevel(zapcore.InfoLevel)
	cfg.DisableCaller = true
	cfg.OutputPaths = []string{"/tmp/non_existing_dir/file.log"}
	logger, err := cfg.Build() // panic()
	if err != nil {
		log.Fatal(err)
	}
	logger.Info("test message", zap.String("foo", "bar"))
}
```

After run: 

```panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x605933]

goroutine 1 [running]:
go.uber.org/zap.Config.openSinks(0xc420011218, 0x100, 0xc420011230, 0x699642, 0x4, 0x699426, 0x3, 0x6998f4, 0x5, 0x699318, ...)
	/usr/src/go/src/go.uber.org/zap/config.go:220 +0x193
go.uber.org/zap.Config.Build(0xc420011218, 0x100, 0xc420011230, 0x699642, 0x4, 0x699426, 0x3, 0x6998f4, 0x5, 0x699318, ...)
	/usr/src/go/src/go.uber.org/zap/config.go:161 +0xb1
main.main()
```

The problem is in `go.uber.org/zap/config.go#220 and 225-226` in openSinks method - there should be check for nil clouseOut/closeErr or zap.Open should return nopcloser function.

**Comment by aajtodd** [comment]
Just ran into this as well when testing on file/directory without enough permission to create the log file.

**Comment by akshayjshah** [comment]
Thanks for the bug report! This should is fixed in #394. Once that lands, I'll cut a new release.
